ui/core-react/src/core-react/utils/hooks/useCrossOriginPopup.tsx

```diff
@@ -0,0 +1,100 @@
+
+  // ONLY re-render the popup when visibility changes, any other changes get deferred until next visibility change
+  // hence the massive use of 'useRef'
+  const popupWindow = React.useRef<Window>();
+  const savedUrl = React.useRef(url);
+  const savedTitle = React.useRef(title);
+  const savedWidth = React.useRef(width);
+  const savedHeight = React.useRef(height);
+  const savedOnClose = React.useRef(onClose);
+
+  React.useEffect(() => { savedUrl.current = url; }, [url]);
+  React.useEffect(() => { savedTitle.current = title; }, [title]);
+  React.useEffect(() => { savedWidth.current = width; }, [width]);
+  React.useEffect(() => { savedHeight.current = height; }, [height]);
+  React.useEffect(() => { savedOnClose.current = onClose; }, [onClose]);
+
+  // Cleanup method after a popup closure.  Also calls the OnClose callback.
+  const handleClosedPopup = React.useCallback(() => {
+    savedOnClose.current();
+    setCheckPopupAliveDelay(undefined);
+    popupWindow.current = undefined;
+  }, []);
+
+  const closePopup = React.useCallback(() => {
+    if (popupWindow.current !== undefined) {
+      popupWindow.current.close();    // Manually close the popup
+      handleClosedPopup();
+    }
+  }, [handleClosedPopup]);
+
+  const checkPopupClosed = React.useCallback(() => {
+    if (popupWindow.current?.closed) {
+      // Popup has been closed by end-user, inform our host and cleanup
+      handleClosedPopup();
+    }
+  }, [handleClosedPopup]);
+
+  //  Close popup when parent window get closed
+  React.useEffect(() => {
+    window.onbeforeunload = (_event) => {closePopup();};
+    return () => {
+      window.onbeforeunload = null;
+    };
+  }, [closePopup]);
+
+  // Whenever the hook is unloaded, make sure the underlying popup get closed.
+  // Note: An interval is used to check if popup was closed by user: because we access
+  // a cross domain resource inside the popup, we don't have access to popup window events.
+  // As a workaround, we periodically check if popup is still alive.
+  // Reference: https://stackoverflow.com/questions/9388380/capture-the-close-event-of-popup-window-in-javascript/48240128#48240128
+  React.useEffect(() => {
+    return () => {
+      closePopup();
+    };
+  }, [closePopup]);
+
+  // Timer that checks if popup was closed by end-user
+  useInterval(checkPopupClosed, checkPopupAliveDelay);
+
+  // ==> Main render effect
+  // Monitors visibility changes and open/close the popup accordingly.
+  React.useEffect(() => {
+    // If visible and a popup window is not already open, open a new popup window
+    if (visible && popupWindow.current === undefined) {
+      const popup = window.open(savedUrl.current, savedTitle.current, `width=${savedWidth.current},height=${savedHeight.current}`);
+      if (popup) {
+        popup.focus();
+        popupWindow.current = popup;
+
+        // Start and interval that will check if popup got closed by user
```

Comment:
Could you hook up a popup.onbeforeunload event listener to avoid timer?

Reply:
As explained in the code comment above, when the popup window is accessing cross-origin resources, the parent window no longer have access to popup events.  
If I could, I would remove the timer too.